### PR TITLE
Add xkey proposal

### PR DESCRIPTION
**Pull request by survived**

### CODE REVIEW

**Comment by tkuhrt** [comment]
@survived : Would you like to discuss this proposal at this Thursday's TOC call (June 13 @ 7:00 AM US/Pacific)? Or should we schedule this for next week's call (June 20 @ 7:00 AM US/Pacific)?

File: HIPs/incubation/xkey.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+---
+layout: default
+title: Xkey
+parent: Hyperledger Improvement Proposals
```
Comment (tkuhrt) [inline HIPs/incubation/xkey.md]:
```suggestion
parent: Incubation
grand_parent: Hyperledger Improvement Proposals
```

File: HIPs/incubation/xkey.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  secret sharing, and more.
+* Applied libraries like fast Paillier encryption, using CRT to speed up (de)encryption
+  and homomorphic operations, among others.
+
+The ultimate goal of this proposal is to make developing and maintaining TSS/MPC protocols 
+easier. We want to improve their reliability, readability, and security because we believe 
+these are the safest and most feature-rich ways to protect and move digital assets.
+
+# Status
+Pending proposal
```
Comment (tkuhrt) [inline HIPs/incubation/xkey.md]:
```suggestion
Proposed for Incubation
```

File: HIPs/incubation/xkey.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+[`cggmp21`]: https://github.com/dfns/cggmp21
+
+[cggmp21-paper]: https://eprint.iacr.org/2021/060 
+[frost-draft]: https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-frost-15
+
+[slip10-spec]: https://github.com/satoshilabs/slips/blob/master/slip-0010.md
+[bip32-spec]: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+
+All our work listed above is currently licensed under dual MIT or Apache-2.0 licenses. 
+We're planning on relicensing the code to Apache-2.0 only.
```
Comment (tkuhrt) [inline HIPs/incubation/xkey.md:110]: Do we need to relicense?

**Review by tkuhrt — CHANGES_REQUESTED** [review]

File: HIPs/incubation/xkey.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+understand, and their implementation adds more layers of complexity. TSS protocols are 
+usually multi-round and require synchronous communication with authenticated and encrypted 
+channels. Their state progresses gradually as they receive messages. Therefore, implementing 
+TSS involves intricate boilerplate code, making it difficult to audit.
+
+# Dependent Projects
+None
+
+# Motivation
+When we implemented a TSS ECDSA protocol, we noticed a lack of tools and frameworks available
```
Comment (yacovm) [inline HIPs/incubation/xkey.md:45]: There is https://github.com/IBM/TSS for Golang.

File: HIPs/incubation/xkey.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+The ultimate goal of this proposal is to make developing and maintaining TSS/MPC protocols 
+easier. We want to improve their reliability, readability, and security because we believe 
+these are the safest and most feature-rich ways to protect and move digital assets.
+
+# Status
+Pending proposal
+
+# Solution
+We developed following crates in Rust:
```
Comment (yacovm) [inline HIPs/incubation/xkey.md:69]: any plans on supporting Golang?

Comment (adecaro) [reply HIPs/incubation/xkey.md:69]: even wrapper would be very helpful.

**Comment by survived** [comment]
> Would you like to discuss this proposal at this Thursday's TOC call (June 13 @ 7:00 AM US/Pacific)? Or should we schedule this for next week's call (June 20 @ 7:00 AM US/Pacific)?

@tkuhrt I can the next week!

File: HIPs/incubation/xkey.md (hunk 5 above)
Comment (survived) [reply HIPs/incubation/xkey.md:69]: Not with our resources, but if somebody from community comes with such contribution (and strong intention to keep developing and maintaining it), I think we'd be happy to accept it!

File: HIPs/incubation/xkey.md (hunk 3 above)
Comment (survived) [reply HIPs/incubation/xkey.md:110]: Hart said it's a strong requirement and a good thing to do in any case, for sake of avoiding future problems related to the licensing and contributions

**Review by petermetz — APPROVED** [review]

**Review by petermetz — COMMENTED** [review]
@survived Thank you for submitting this! Based on what I read it looks like a good addition so far!
Quick question: Is there an existing repository out there somewhere to go and look at the code or do you plan on opening up the sources on condition of the proposal being accepted? I'd love to have a sneak preview at the code itself.

Edit: Nvm, I found the sources on GitHub: https://github.com/dfns

**Comment by survived** [comment]
Hi @petermetz, there's a "Solution" section in the HIP which contains all the links to existing repos we want to contribute as the project

**Review by jimthematrix — APPROVED** [review]
Sounds like a useful collection of libraries that performs a range of cryptography. Some of the library descriptions, such as the 3rd one, sound very generic with `provides ZK proofs and tools`.

**Review by yacovm — APPROVED** [review]

**Review by VRamakrishna — APPROVED** [review]

**Comment by yacovm** [comment]
> Sounds like a useful collection of libraries that performs a range of cryptography. Some of the library descriptions, such as the 3rd one, sound very generic with `provides ZK proofs and tools`.

I think the ZKP they refer to, are not generic as in "generic computation" but rather generic as in "common" sub-protocols to be used as part of the MPC protocol.

**Comment by ryjones** [comment]
Approved on 20 JUN 2024

**Comment by survived** [comment]
> I think the ZKP they refer to, are not generic as in "generic computation" but rather generic as in "common" sub-protocols to be used as part of the MPC protocol.

Yes exactly. `generic-ec` is a library for generic elliptic curve cryptography, and `generic-ec-zkp` is a library with ZK proofs on top of `generic-ec` lib (thus the `-zkp` suffix).

**Review by tkuhrt — APPROVED** [review]

**Pull request merged**